### PR TITLE
Optimized rebuild method

### PR DESCRIPTION
**Pull request by ipakeev**
Rebuilds trees 5-10 times faster.

### CODE REVIEW

**Review by matthiask — COMMENTED** [review]
Thanks! I didn't read the code too closely but it does look like it does the right thing.

Is there a reason to add a whole new manager method though? Can't you modify the existing rebuild functionality instead? This would also mean that existing tests (if there are any) can be reused.

**Comment by ipakeev** [comment]
Done 👍. 
I also optimized the partial_rebuild method.

File: mptt/managers.py [hunk 1]
```diff
@@ -619,48 +620,102 @@ def root_nodes(self):
...
             qs = qs.order_by(*opts.order_insertion_by)
-        pks = qs.values_list("pk", flat=True)
 
-        rebuild_helper = self._rebuild_helper
-        idx = 0
-        for pk in pks:
-            idx += 1
-            rebuild_helper(pk, 1, idx)
+        children = defaultdict(list)
+        for child in qs.only("id", "parent_id"):
```
Comment (matthiask) [inline mptt/managers.py:647]: I'm wondering. Is it guaranteed that those two fields are named `id` and `parent_id`? Shouldn't you use `pk` and `_mptt_meta.parent_attr` or something?

**Review by matthiask — COMMENTED** [review]
Thanks, this is looking good! I have a doubt re. the naming of model fields but apart from that I think it's good to go. Nice to see the improvements in the testsuite.

File: mptt/managers.py (hunk 1 above)
Comment (ipakeev) [reply mptt/managers.py]: Yes, that's a good idea.

**Comment by matthiask** [comment]
Thank you!

File: mptt/managers.py [hunk 2]
```diff
@@ -619,48 +620,102 @@ def root_nodes(self):
...
             qs = qs.order_by(*opts.order_insertion_by)
-        pks = qs.values_list("pk", flat=True)
 
-        rebuild_helper = self._rebuild_helper
-        idx = 0
-        for pk in pks:
-            idx += 1
-            rebuild_helper(pk, 1, idx)
+        children = defaultdict(list)
+        for child in qs.select_related("parent"):
```
Comment (henrikek) [inline mptt/managers.py:647]: When I upgrade to version 0.15 i run into problem when the name of related is not "parent". Why not use opts.parent_attr?

**Pull request merged**